irods-vfs-impl/src/main/java/org/irods/nfsrods/vfs/IRODSVirtualFileSystem.java

```diff
@@ -1645,15 +1663,101 @@ private static Inode toFh(long _inodeNumber)
         return Inode.forFile(Longs.toByteArray(_inodeNumber));
     }
 
+    private String tryAndFindPathFromGenQuery(long _inodeNumber, String option) {
```

Comment:
Consider changing the return type to `Optional<String>`.

That improves the interface by allowing the next developer to quickly know that the method **may** or **may not** return a valid value.